apps/web/src/app/(private)/settings/_components/ProviderApiKeys/New/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { Providers } from '@latitude-data/core/browser'
+import {
+  Button,
+  FormWrapper,
+  Input,
+  Modal,
+  Select,
+} from '@latitude-data/web-ui'
+import useProviderApiKeys from '$/stores/providerApiKeys'
+
+export default function NewApiKey({
+  setOpen,
+}: {
+  setOpen: (open: boolean) => void
+}) {
+  const { create } = useProviderApiKeys()
+  const createAction = async (formData: FormData) => {
+    try {
+      await create({
+        name: formData.get('name') as string,
+        provider: formData.get('provider') as Providers,
+        token: formData.get('token') as string,
+      })
+
+      setOpen(false)
+    } catch (_) {
+      // do nothing
+    }
+  }
+  const onOpenChange = (open: boolean) => {
+    setTimeout(() => setOpen(open), 100)
```

Comment:
Why this?

Reply:
it's a better ux (show the modal fading out instead of hard closing)

Reply:
This should be cooked in default behavior of the modal. Animation is not give time to display CSS animation

Reply:
Don't use `defaultOpen`. Use `open` prop and don't conditionally render the modal